src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java

```diff
@@ -1494,15 +1509,19 @@ public boolean canSee(Player player)
 	@Override
 	public boolean isFlying()
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		return flying;
 	}
 
 	@Override
 	public void setFlying(boolean value)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		this.flying = value;
+	}
+
+	public void simulateToggleFlight(boolean fly)
+	{
+		PlayerToggleFlightEvent event = new PlayerToggleFlightEvent(this, fly);
+		Bukkit.getPluginManager().callEvent(event);
```

Comment:
The method should respect the Event outcome and set the `flying` state accordingly.